pkg/reconciler/common/transformers.go

```diff
@@ -940,3 +940,49 @@ func ReplaceNamespace(newNamespace string) mf.Transformer {
 		return nil
 	}
 }
+
+// AddSecretData adds the given data and annotations to the Secret object.
+func AddSecretData(data map[string][]byte, annotations map[string]string) func(*unstructured.Unstructured) error {
+	return func(u *unstructured.Unstructured) error {
+		// If input data is empty, do not transform
+		if len(data) == 0 {
+			return nil
+		}
+
+		// Check if the resource is a Secret
+		if u.GetKind() != "Secret" {
+			return nil
+		}
+
+		// Convert unstructured to Secret
+		secret := &corev1.Secret{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, secret)
+		if err != nil {
+			return err
+		}
+
+		// Update the Secret's data only if it is nil or empty
+		if secret.Data == nil || len(secret.Data) == 0 {
```

Comment:
@jkhelil I believe `len` should return `0` on `nil` value too. If so, we can remove the `nil` check